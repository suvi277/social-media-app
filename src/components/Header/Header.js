import React from 'react';
import './Header.scss';
import { Logo } from '@app/components/Logo';
import { Navigation } from '@app/components/Navigation';

const Header = (props) => {
	return (
		<header className="sticky-top navbar p-0">
			<div className="container">
				<Logo />
				<Navigation {...props} />
			</div>
		</header>
	);
};

export { Header };
