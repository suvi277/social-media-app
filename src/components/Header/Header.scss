header {
    background-color: #263238;
    top: 0;
    width: 100%;
    width: 100%;
    position: sticky;
}