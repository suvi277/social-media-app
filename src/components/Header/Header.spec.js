import React from 'react';
import { shallow } from 'enzyme';
import { Header } from './Header';

describe('Header Component', () => {
	const props = {
		someProps: 'someprops'
	};
	const wrapper = shallow(<Header {...props} />);

	it('renders as expected', () => {
		expect(wrapper).toMatchSnapshot();
	});

	it('should exist Logo Component', () => {
		expect(wrapper.find('Logo')).toHaveLength(1);
	});

	it('should exist Header Component', () => {
		expect(wrapper.find('Navigation')).toHaveLength(1);
	});

	it('should render Header Component with Header props', () => {
		const navigation = wrapper.find('Navigation');
		expect(navigation.props().someProps).toBe(props.someProps);
	});
});
