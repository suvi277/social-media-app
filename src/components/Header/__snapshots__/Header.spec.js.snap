// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component renders as expected 1`] = `ShallowWrapper {}`;
